### PR TITLE
🏗 Ignore markdown templates in link checker by convention

### PR DESCRIPTION
**Pull request by alanorozco**
This might not be necessary if the ignore-by-backtick strategy works.

@rsimha Unsure how it would work with actual link syntax:

```md
[${name} rules](https://github.com/ampproject/amphtml/blob/master/extensions/${name}/validator-${name}.protoascii) 
```

Also some modernizing and cleanup.

(Offshoot of #25164)

### CODE REVIEW

**Comment by rsimha** [comment]
> @rsimha Unsure how it would work with actual link syntax:
> 
> ```lisp
> [${name} rules](https://github.com/ampproject/amphtml/blob/master/extensions/${name}/validator-${name}.protoascii) 
> ```

I tried a few things locally and realized that filtering out templated links is the way to go. Actual review of this PR coming up soon.

File: build-system/tasks/check-links.js [hunk 1]
```diff
@@ -51,68 +51,51 @@ function getMarkdownFiles() {
...
-  });
-  return BBPromise.all(linkCheckers).then(function(allResults) {
-    let deadLinksFound = false;
-    const filesWithDeadLinks = [];
-    allResults.map(function(results, index) {
-      // Skip files that were deleted by the PR.
-      if (!fs.existsSync(markdownFiles[index])) {
-        return;
-      }
+  const allResults = await Promise.all(markdownFiles.map(runLinkChecker));
```
Comment (rsimha) [inline build-system/tasks/check-links.js]: I now see what you said about this cleanup not materially change the concurrency. Can you confirm by testing locally as follows? (Logs should be separated by file.)
```
gulp check-links --files "contributing/DEVELOPING.md,contributing/getting-started-e2e.md,contributing/getting-started-quick.md"
```

File: build-system/tasks/check-links.js [hunk 2]
```diff
@@ -128,12 +111,12 @@ function isLinkToFileIntroducedByPR(link) {
...
 /**
- * Filters out whitelisted links before running the link checker.
+ * Filters out links in allow-list before running the link checker.
  *
  * @param {string} markdown Original markdown.
- * @return {string} Markdown after filtering out whitelisted links.
+ * @return {string} Markdown after filtering out allowed links.
  */
-function filterWhitelistedLinks(markdown) {
+function filterAllowedLinks(markdown) {
```
Comment (rsimha) [inline build-system/tasks/check-links.js]: The latest [API docs](https://github.com/tcort/markdown-link-check#markdownlinkcheckmarkdown-opts-callback) of the link checker appear to have introduced an `ignorePatterns` option that lets you specify a list of regexes to ignore. I think it's cleaner than modifying the markdown text before passing it to the checker. Will do this in a follow up PR. Nothing to do now.

File: build-system/tasks/check-links.js (hunk 1 above)
Comment (alanorozco) [reply build-system/tasks/check-links.js]: [LGTM](https://gist.github.com/alanorozco/92c889bd51868d44398df9606f625fb6)

File: build-system/tasks/check-links.js (hunk 2 above)
Comment (rsimha) [reply build-system/tasks/check-links.js]: Done in #25235.

**Pull request merged**